linux/utils/create_disk_partition.yml

```diff
@@ -0,0 +1,104 @@
+# Copyright 2022 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Description:
+#   Format and create 1 new partition on a disk
+# Parameters:
+#   disk_name: The disk name to be formatted and create partition
+#   partition_fstype: Create filesystem on new partition, which could be ext4, xfs
+#                         or btrfs, etc. The default filesystem is ext4.
+# Return:
+#   partition_name: The disk partition name
+#   partition_device_path: The disk partition device path
+#
+- name: "Initialize disk partition name and device path"
+  ansible.builtin.set_fact:
+    partition_name: ""
+    partition_device_path: ""
+
+- name: "Set partition filesystem type"
+  block:
+    - name: "Check filesystem is supported by guest OS"
+      ansible.builtin.shell: "which mkfs.{{ partition_fstype }}"
+      ignore_errors: true
+      changed_when: false
+      delegate_to: "{{ vm_guest_ip }}"
+      register: check_fs
+
+    - name: "Filesystem {{ partition_fstype }} is not supported, change partition filesystem to ext4"
+      ansible.builtin.set_fact:
+        partition_fstype: "ext4"
+      when: >
+        check_fs is undefined or
+        check_fs.rc is undefined or
+        check_fs.rc != 0
+  when:
+    - partition_fstype is defined
+    - partition_fstype
+
+- name: "Set default partition filesystem to ext4"
```

Comment:
Should this default value be put at the beginning and failed task when `partition_fstype` not found?